src/main/java/org/ibp/api/rest/labelprinting/GermplasmLabelPrinting.java

```diff
@@ -302,169 +303,146 @@ void setAddedColumnsToSearchRequest(
-				return;
+				columns.put(combinedKey, Objects.toString(germplasmSearchResponse.getGermplasmUUID(), ""));
+				break;
 			case REFERENCE:
-				columns.put(key, Objects.toString(germplasmSearchResponse.getReference(), ""));
-				return;
+				columns.put(combinedKey, Objects.toString(germplasmSearchResponse.getReference(), ""));
+				break;
 			case METHOD_CODE:
-				columns.put(key, Objects.toString(germplasmSearchResponse.getMethodCode(), ""));
-				return;
+				columns.put(combinedKey, Objects.toString(germplasmSearchResponse.getMethodCode(), ""));
+				break;
 			case METHOD_NUMBER:
-				columns.put(key, Objects.toString(germplasmSearchResponse.getMethodNumber(), ""));
-				return;
+				columns.put(combinedKey, Objects.toString(germplasmSearchResponse.getMethodNumber(), ""));
+				break;
 			case METHOD_GROUP:
-				columns.put(key, Objects.toString(germplasmSearchResponse.getMethodGroup(), ""));
-				return;
+				columns.put(combinedKey, Objects.toString(germplasmSearchResponse.getMethodGroup(), ""));
+				break;
 			case GROUP_SOURCE_GID:
-				columns.put(key, Objects.toString(germplasmSearchResponse.getGroupSourceGID(), ""));
-				return;
+				columns.put(combinedKey, Objects.toString(germplasmSearchResponse.getGroupSourceGID(), ""));
+				break;
 			case GROUP_SOURCE_PREFERRED_NAME:
-				columns.put(key, Objects.toString(germplasmSearchResponse.getGroupSourcePreferredName(), ""));
-				return;
+				columns.put(combinedKey, Objects.toString(germplasmSearchResponse.getGroupSourcePreferredName(), ""));
+				break;
 			case LOTS:
-				columns.put(key, Objects.toString(germplasmSearchResponse.getLotCount(), ""));
-				return;
+				columns.put(combinedKey, Objects.toString(germplasmSearchResponse.getLotCount(), ""));
+				break;
+			case CROSS:
+				columns.put(combinedKey, Objects.toString(germplasmSearchResponse.getPedigreeString(), ""));
+				break;
+			case IMMEDIATE_SOURCE_GID:
+				columns.put(combinedKey, Objects.toString(germplasmSearchResponse.getImmediateSourceGID(), ""));
+				break;
+			case IMMEDIATE_SOURCE_NAME:
+				columns.put(combinedKey, Objects.toString(germplasmSearchResponse.getImmediateSourceName(), ""));
+				break;
 			default:
 				//do nothing
 		}
 	}
 
+	void getNameDataRowValue(final Map<String, String> columns, final boolean isPdf, final String combinedKey,
+		final GermplasmSearchResponse germplasmSearchResponse, final Map<Integer, Map<Integer, String>> nameValues){
+		final Map<Integer, String> namesByType = nameValues.get(germplasmSearchResponse.getGid());
+		if (namesByType != null) {
+			final String nameValue = namesByType.get(this.getFieldIdFromCombinedKey(combinedKey));
+			if (nameValue != null) {
+				columns.put(combinedKey, this.truncateValueIfPdf(isPdf, nameValue, GermplasmLabelPrinting.NAME_DISPLAY_MAX_LENGTH));
+			}
+		}
+	}
+
+	private String truncateValueIfPdf(final boolean isPdf, final String value, final int maxLength) {
```

Comment:
@cuenyad I would rather having truncateValue and make the isPdf comparisson out of the function, not recommended to send a boolean to a function and perform actions based on it

Reply:
I created the [IBP-6111](https://ibplatform.atlassian.net/browse/IBP-6111) to continue with this change.